numba/targets/builtins.py

```diff
@@ -1324,6 +1312,96 @@ def complex_impl(context, builder, sig, args):
     res = cmplx._getvalue()
     return impl_ret_untracked(context, builder, sig.return_type, res)
 
+
+@builtin
+@implement(types.NumberClass, types.Any)
+def number_constructor(context, builder, sig, args):
+    """
+    Convert any number to any other.
+    """
+    [val] = args
+    [valty] = sig.args
+    return context.cast(builder, val, valty, sig.return_type)
+
+
+#-------------------------------------------------------------------------------
+# Implicit casts between numerics
```

Comment:
Le 30/12/2015 19:43, stuartarchibald a écrit :
> 
> Are these implicit cast combinations exhaustively tested somewhere?

There's test_numconv and test_numberctor, yes.